pkg/types/validation/installconfig.go

```diff
@@ -464,3 +465,33 @@ var (
 		return v
 	}()
 )
+
+func validateCloudCredentialsMode(mode types.CloudCredentialsMode, fldPath *field.Path, platform string) field.ErrorList {
+	if mode == "" {
+		return nil
+	}
+	validModes := map[types.CloudCredentialsMode]bool{}
+	switch platform {
+	case aws.Name, azure.Name, gcp.Name:
+		validModes[types.MintCloudCredentialsMode] = true
+		validModes[types.PassthroughCloudCredentialsMode] = true
+	}
+	switch platform {
+	case aws.Name:
+		validModes[types.ManualCloudCredentialsMode] = true
+	}
+	if validModes[mode] {
+		return nil
+	}
+	allErrs := field.ErrorList{}
+	if len(validModes) != 0 {
+		validModeValues := make([]string, 0, len(validModes))
+		for m := range validModes {
+			validModeValues = append(validModeValues, string(m))
+		}
+		allErrs = append(allErrs, field.NotSupported(fldPath, mode, validModeValues))
+	} else {
+		allErrs = append(allErrs, field.Invalid(fldPath, mode, fmt.Sprintf("cloud credentials mode cannot be set when using the %q platform", platform)))
```

Comment:
the invalid error already includes the field name not i do not think we need to repeat `cloud credentials mode cannot` ..